modules/pipeline/providers/reconciler/interface.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reconciler
+
+import (
+	"context"
+
+	"github.com/erda-project/erda/modules/pipeline/providers/reconciler/legacy/reconciler"
+)
+
+type Interface interface {
+	Reconcile(ctx context.Context, pipelineID uint64)
+
+	// TODO use provider init
+	InjectLegacyReconciler(r *reconciler.Reconciler)
+}
+
+func (p *provider) Reconcile(ctx context.Context, pipelineID uint64) {
+	p.r.Reconcile(ctx, pipelineID)
+}
+
+func (p *provider) InjectLegacyReconciler(r *reconciler.Reconciler) {
+	p.r = r
+}
```

Comment:
why place impl func into interface.go?

Reply:
legacy reconciler coupled with some services, can not be initialized directly now.
after legacy reconciler refactored, remove this inject method.